### PR TITLE
[e2e] tag resources by teamTag from config

### PR DESCRIPTION
**Pull request by pducolin**
### What does this PR do?

Parse `teamTag` from config file and use it to tag resources. 

### Motivation

This PR adds team tag to cloud resources, helping monitoring cost and test adoption

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: test/new-e2e/runner/parameters/fixtures/config_with_stackparams.yaml [hunk 1]
```diff
@@ -11,6 +11,8 @@ configParams:
     # use the public key you registered on aws account
     # required to configure Windows EC2 instances
     publicKeyPath: "/Users/totoro/.ssh/id_rsa.pub"
+    # team tag labeling all resources
```
[the remark this answers is not in the document]
Comment (pducolin) [reply test/new-e2e/runner/parameters/fixtures/config_with_stackparams.yaml]: I updated comment in the example and removed them from the fixtures here, as they were misleading. Updates for the example https://github.com/DataDog/test-infra-definitions/pull/247

Should we have something here that points to how to create a local config profile ? Note it is not mandatory, it works with env variables too.

[the remark this answers is not in the document]
Comment (pducolin) [reply test/new-e2e/runner/parameters/fixtures/config_with_stackparams.yaml]: I think, not sure if we fail here if there is no config file

File: test/new-e2e/runner/parameters/fixtures/config_no_aws_account.yaml [hunk 2]
```diff
@@ -1,21 +1,10 @@
-# Environments configuration, validated and passed to Pulumi
```
Comment (ogaca-dd) [inline test/new-e2e/runner/parameters/fixtures/config_no_aws_account.yaml:1]: What do you think about adding `test` in the filename to make it explicit they are used in test?

**Review by ogaca-dd — COMMENTED** [review]
Can you update `.test_infra_config.yaml.example`?

**Comment by pducolin** [comment]
> Can you update .test_infra_config.yaml.example?
Addressed in https://github.com/DataDog/test-infra-definitions/pull/247/commits/e43a88ca4ab0893a024a6af8b94124e9043eeab2

**Review by ogaca-dd — APPROVED** [review]

**Pull request merged**